clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -9290,6 +9290,8 @@ def warn_unused_result_typedef_unsupported_spelling : Warning<
 def warn_unused_volatile : Warning<
   "expression result unused; assign into a variable to force a volatile load">,
   InGroup<DiagGroup<"unused-volatile-lvalue">>;
+def note_nodiscard_specified_here : Note<
```

Comment:
this should probably just re-use an existing note, perhaps the one we have for deprecated?

Reply:
That note currently reads:
```
def note_availability_specified_here : Note<
  "%0 has been explicitly marked "
  "%select{unavailable|deleted|deprecated}1 here">;
```
Currently, this is the combined note used for unavailable, deleted, and deprecated, and the flavor is chosen by a number (0/1/2) passed as the second argument. Probably, if we want to reuse this, that should be changed to a string parameter?

Reply:
Either a string parameter, or add another select entry/ies.  Thats easy enough. An alternative is to add an enum for them with an operator << overload available.

Reply:
Still want this combined/generalized.  These are incredibly similar in both use and content.